### PR TITLE
Add guests to `WedLog`

### LINKED ISSUE #43: Add `guests` field to AddressBook
The application lacks support for maintaining guests. As such, we need to
* Modify `AddressBook` to include `UniqueGuestList` that stores guests.
* Add relevant storage logic

This issue was originally #27.

### PR DESCRIPTION
**Pull request by p-xp**
Resolves #43.

Our application does not maintain guests in the system.

Storing guest information is a key functionality that users will
expect from a wedding planning app.

Let's add a guest list to model.AddressBook, creating the relevant
classes and methods to save, load and edit guests.

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedGuest.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.RsvpStatus;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Jackson-friendly version of {@link Guest}.
+ */
+class JsonAdaptedGuest {
```
Comment (samuelim01) [inline src/main/java/seedu/address/storage/JsonAdaptedGuest.java:25]: Perhaps this class can extend from JsonAdaptedPerson?

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 2]
```diff
@@ -54,6 +60,13 @@ public AddressBook toModelType() throws IllegalValueException {
             }
             addressBook.addPerson(person);
         }
+        for (JsonAdaptedGuest jsonAdaptedGuest : guests) {
+            Guest guest = jsonAdaptedGuest.toModelType();
+            if (addressBook.hasGuest(guest)) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_GUESTS);
+            }
+            addressBook.addGuest(guest);
+        }
```
Comment (samuelim01) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:69]: Could you add some test methods for this?

**Review by samuelim01 — COMMENTED** [review]
Overall, LGTM

File: src/main/java/seedu/address/model/AddressBook.java [hunk 3]
```diff
@@ -30,6 +34,7 @@ public class AddressBook implements ReadOnlyAddressBook {
      */
     {
         persons = new UniquePersonList();
```
Comment (keaganpzh) [inline src/main/java/seedu/address/model/AddressBook.java:36]: Are we gonna keep track of `persons` as well?

**Review by keaganpzh — COMMENTED** [review]
Is there a need to keep track of a list of `Person`?

**Review by keaganpzh — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/model/AddressBook.java (hunk 3 above)
Comment (samuelim01) [reply src/main/java/seedu/address/model/AddressBook.java:36]: Perhaps we can discuss our overall strategy on how and when we we will eliminate Person / AddressBook commands from the repo?

**Pull request merged**